django/forms/widgets.py

```diff
@@ -588,11 +589,15 @@ def optgroups(self, name, value, attrs=None):
                 group_name = option_value
                 subindex = 0
                 choices = option_label
+                groups.append((group_name, subgroup, index))
             else:
-                group_name = None
+                if subgroup_None is None:
+                    subgroup_None = []
+                    groups.append((None, subgroup_None, index))
```

Comment:
This is going to change the ordering of my choices. 

Take the example from the test: 

```
 + choices = [
 +            ("one", 1),
 +            [
 +                "italian", (
 +                    ("uno", 1),
 +                    ("due", 2)
 +                )
 +            ],
 +            ("two", 2),
 +        ]
```

I may deliberately want to present the choices in that order. An/some ungrouped default/common choices at the beginning, some groups, and then some ungrouped other choices at the end. 

I may have some grouped choices, then some ungrouped ones, then some more groups, and so on. With this change all the ungrouped choices would be in the second group, regardless of that ordering. 

I'm not sure we just want to enforce them all being grouped at whatever index the first ungrouped choice appears.